<launch>

  <arg name="task2" default="true"/>

  <group if="$(arg task2)">
    <include file="$(find challenge_project)/launch/challenge3/challenge3_task2.launch"/>
  </group>

  <group unless="$(arg task2)">
    <param name="task1_running" type="bool" value="false"/>
    <include file="$(find challenge_project)/launch/gazebo.launch">
      <arg name="x" value="-18"/>
      <arg name="y" value="3.2"/>
      <arg name="z" value="0.02"/>
      <arg name="roll" value="0"/>
      <arg name="pitch" value="0"/>
      <arg name="yaw" value="3.14"/>
      <arg name="world" value="$(find challenge_project)/worlds/challenge3.world"/>
    </include>
    <param name="task2_part2_running" type="bool" value="false"/>
  <node name="OBSTACLE" pkg="challenge_project" type="challenge3b_world_control.py"/>
  </group>

  <param name="linear_scale" type="double" value="0.5" />
   <node pkg="challenge_project" name="navigation" type="navigation.py" launch-prefix="xterm -e"/>

  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/camera/image_raw" />
  </node>
</launch>
